crates/bevy_ecs/hecs/src/archetype.rs

```diff
@@ -400,6 +400,8 @@ impl Archetype {
         let state = self.state.get_mut(&ty).unwrap();
         if added {
             state.added_entities[index] = true;
+        } else {
```

Comment:
Hey sorry for the delay here. These changes make sense to me. I _think_ this works as-is, but I'm a little worried that the implicit `!added == mutated` here could cause problems / create corner case issues. I'd rather just pass in a `mutated` flag and set it explicitly each time.

Reply:
Im specifically worried about cases where we change archetypes and some of the components aren't mutated.

Reply:
Ok, I did the changes you suggested and I think it's better like this because further call of the method will need to be explicit on the state.